src/context/engagement_context/engagement_hook.tsx

```diff
@@ -0,0 +1,234 @@
+import {
```

Comment:
The logic from `engagement_context` was moved to this hook. There is not significant net new code in this file, just a consolidation of the logic from `engagement_context` and `engagement_form_context`.